action/invoke.go

```diff
@@ -14,7 +14,33 @@ type InvokeRequest struct {
 	// Config is the configuration the user supplied for the action.
 	Config tfsdk.Config
 
-	// TODO:Actions: Add linked resources once lifecycle/linked actions are implemented
+	// LinkedResources contains the data of the managed resource types that are linked to this action.
+	//
+	//   - If the action schema type is Unlinked, this field will be empty.
+	//   - If the action schema type is Lifecycle, this field will contain a single linked resource.
+	//   - If the action schema type is Linked, this field will be one or more linked resources, which
+	//     will be in the same order as the linked resource schemas are defined in the action schema.
+	//
+	// For Lifecycle actions, the provider may only change computed-only attributes of the linked resources.
+	// For Linked actions, the provider may change any attributes of the linked resources.
+	LinkedResources []InvokeRequestLinkedResource
+}
+
+// InvokeRequestLinkedResource represents linked resource data before the action is invoked.
+type InvokeRequestLinkedResource struct {
```

Comment:
The only thing not represented in this request that is normally in the plan, is private data.... Should that be in the protocol?